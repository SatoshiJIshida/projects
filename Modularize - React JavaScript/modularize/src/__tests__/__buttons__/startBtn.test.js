import React from "react";
import { shallow } from "enzyme";
import { render, screen, fireEvent } from "@testing-library/react";
import StartBtn from "../../components/buttons/startBtn";

/**
 * Library Jest taken from source https://jestjs.io/
 * Library Enzyme taken from source https://www.npmjs.com/package/enzyme
 * Library React Testing Library taken from source https://testing-library.com/docs/react-testing-library/intro/
 */

describe("Button that starts the game and takes user to the Intro Screen", () => {
  it("should be defined", () => {
    expect(StartBtn).toBeDefined();
  });

  it("should render correctly", () => {
    const button = shallow(<StartBtn name="button-test" />);
    expect(button).toMatchSnapshot();
  });

  it("should call mock function when clicked", () => {
    const handleClick = jest.fn();
    render(<StartBtn onClick={handleClick} />);
    fireEvent.click(screen.getByText(/Start/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
