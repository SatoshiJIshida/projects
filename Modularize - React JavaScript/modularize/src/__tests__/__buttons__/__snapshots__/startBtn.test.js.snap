// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button that starts the game and takes user to the Intro Screen should render correctly 1`] = `ShallowWrapper {}`;
